Detectors/EMCAL/calibration/run/runCalibOffline.cxx

```diff
@@ -187,22 +193,29 @@ int main(int argc, char** argv)
     o2::emcal::BadChannelMap BCMap;
 
     BCMap = CalibExtractor.calibrateBadChannels(hCalibInputHist);
-
     // store bad channel map in ccdb via emcal calibdb
-    std::map<std::string, std::string> metadata;
-    calibdb.storeBadChannelMap(&BCMap, metadata, rangestart, rangeend);
+    if (doLocal) {
+      std::unique_ptr<TFile> writer(TFile::Open("bcm.root", "RECREATE"));
+      writer->WriteObjectAny(&BCMap, "o2::emcal::BadChannelMap", "BadChannelMap");
+    } else {
+      std::map<std::string, std::string> metadata;
+      calibdb.storeBadChannelMap(&BCMap, metadata, rangestart, rangeend);
+    }
+
   } else {
     printf("perform time calibration analysis\n");
 
     // calibrate the time
     o2::emcal::TimeCalibrationParams TCparams;
     TCparams = CalibExtractor.calibrateTime(hCalibInputHist, timeRangeLow, timeRangeHigh);
 
-    // store parameters in ccdb via emcal calibdb
-    std::map<std::string, std::string> metadata;
-    calibdb.storeTimeCalibParam(&TCparams, metadata, rangestart, rangeend);
+    if (!doLocal) {
+      // store parameters in ccdb via emcal calibdb
+      std::map<std::string, std::string> metadata;
+      calibdb.storeTimeCalibParam(&TCparams, metadata, rangestart, rangeend);
+    }
 
-    if (namePathStoreLocal.find(".root") != std::string::npos) {
+    if (namePathStoreLocal.find(".root") != std::string::npos && doLocal) {
```

Comment:
For the time calib this is actually no the `local` mode. Recall: For the local mode we export the calibration object to a file which can then be uploaded manually to the CCDB. What was implemented for the time calib however is to export a ROOT hist to file (for visualization), which is not the calibration object. Like this we cannot post the time calib object to the CCDB. I suggest to implement like for the bad channel calib, because once the object is stream to a ROOT file it is trivial to extract from this the ROOT histogram for visualization.

Reply:
Thanks for the comments, I have fixed this and re-pushed!